libs/core/langchain_core/beta/runnables/context.py

```diff
@@ -1,401 +0,0 @@
-    steps: list[Runnable],
-    setter: Callable,
-    getter: Callable,
-    event_cls: Union[type[threading.Event], type[asyncio.Event]],
-) -> RunnableConfig:
-    if any(k.startswith(CONTEXT_CONFIG_PREFIX) for k in config.get("configurable", {})):
-        return config
-
-    context_specs = [
-        (spec, i)
-        for i, step in enumerate(steps)
-        for spec in step.config_specs
-        if spec.id.startswith(CONTEXT_CONFIG_PREFIX)
-    ]
-    grouped_by_key = {
-        key: list(group)
-        for key, group in groupby(
-            sorted(context_specs, key=lambda s: s[0].id),
-            key=lambda s: _key_from_id(s[0].id),
-        )
-    }
-    deps_by_key = {
-        key: {
-            _key_from_id(dep) for spec in group for dep in (spec[0].dependencies or [])
-        }
-        for key, group in grouped_by_key.items()
-    }
-
-    values: Values = {}
-    events: defaultdict[str, Union[asyncio.Event, threading.Event]] = defaultdict(
-        event_cls
-    )
-    context_funcs: dict[str, Callable[[], Any]] = {}
-    for key, group in grouped_by_key.items():
-        getters = [s for s in group if s[0].id.endswith(CONTEXT_CONFIG_SUFFIX_GET)]
-        setters = [s for s in group if s[0].id.endswith(CONTEXT_CONFIG_SUFFIX_SET)]
-
-        for dep in deps_by_key[key]:
-            if key in deps_by_key[dep]:
-                msg = f"Deadlock detected between context keys {key} and {dep}"
-                raise ValueError(msg)
-        if len(setters) != 1:
-            msg = f"Expected exactly one setter for context key {key}"
-            raise ValueError(msg)
-        setter_idx = setters[0][1]
-        if any(getter_idx < setter_idx for _, getter_idx in getters):
-            msg = f"Context setter for key {key} must be defined after all getters."
-            raise ValueError(msg)
-
-        if getters:
-            context_funcs[getters[0][0].id] = partial(getter, events[key], values)
-        context_funcs[setters[0][0].id] = partial(setter, events[key], values)
-
-    return patch_config(config, configurable=context_funcs)
-
-
-def aconfig_with_context(
-    config: RunnableConfig,
-    steps: list[Runnable],
-) -> RunnableConfig:
-    """Asynchronously patch a runnable config with context getters and setters.
-
-    Args:
-        config: The runnable config.
-        steps: The runnable steps.
-
-    Returns:
-        The patched runnable config.
-    """
-    return _config_with_context(config, steps, _asetter, _agetter, asyncio.Event)
-
-
-def config_with_context(
-    config: RunnableConfig,
-    steps: list[Runnable],
-) -> RunnableConfig:
-    """Patch a runnable config with context getters and setters.
-
-    Args:
-        config: The runnable config.
-        steps: The runnable steps.
-
-    Returns:
-        The patched runnable config.
-    """
-    return _config_with_context(config, steps, _setter, _getter, threading.Event)
-
-
-@beta()
-class ContextGet(RunnableSerializable):
```

Comment:
out of abundance caution maybe we leave stup class with nice deprecation message on init?